tests/queries/0_stateless/03166_from_readable_size.sql

```diff
@@ -0,0 +1,50 @@
+-- Should be the inverse of formatReadableSize
+SELECT formatReadableSize(fromReadableSize('1 B'));
+SELECT formatReadableSize(fromReadableSize('1 KiB'));
+SELECT formatReadableSize(fromReadableSize('1 MiB'));
+SELECT formatReadableSize(fromReadableSize('1 GiB'));
+SELECT formatReadableSize(fromReadableSize('1 TiB'));
+SELECT formatReadableSize(fromReadableSize('1 PiB'));
+SELECT formatReadableSize(fromReadableSize('1 EiB'));
+
+-- Is case-insensitive
+SELECT formatReadableSize(fromReadableSize('1 mIb'));
+
+-- Should be able to parse decimals
+SELECT fromReadableSize('1.00 KiB');    -- 1024
```

Comment:
We could add tests for floats with
- 0 fractional digits,
- more fractional digits than allowed for Float64.
- any other creative float formats (maybe they can be written in octal, with leading + or -, as NaN etc.)

I would not mind too much about the particular behavior for these special cases, it is more important that these cases are tested and have defined results (e.g. they are successfully parsed or throw an exception).

Reply:
Added several additional tests, I'll add a couple more today as I've found a funky behaviour when input is `'NaN KiB'`